### PR TITLE
MDEV-24966 Galera multi-master regression

### PR DESCRIPTION
**Pull request by sjaakola**
Intermediate commit to enable testing in jenkins

### CODE REVIEW

File: sql/wsrep_thd.cc [hunk 1]
```diff
@@ -348,8 +348,16 @@ bool wsrep_bf_abort(const THD* bf_thd, THD* victim_thd)
 
   if (WSREP(victim_thd) && !victim_thd->wsrep_trx().active())
```
Comment (temeo) [inline sql/wsrep_thd.cc:349]: I'd suggest to protect access to `victim_thd` here by grabbing `LOCK_thd_data`.

File: sql/wsrep_thd.cc [hunk 2]
```diff
@@ -348,8 +348,16 @@ bool wsrep_bf_abort(const THD* bf_thd, THD* victim_thd)
...
+     WSREP_DEBUG("wsrep_bf_abort, BF abort for non active transaction");
+     switch (victim_thd->wsrep_trx().state()) {
+     case wsrep::transaction::s_aborting: /* fall through */
+     case wsrep::transaction::s_aborted:
+       WSREP_DEBUG("victim is aborting or has aborted");
+      return false;
+     default: break;
+     }
+
+     wsrep_start_transaction(victim_thd, victim_thd->wsrep_next_trx_id());
```
Comment (temeo) [inline sql/wsrep_thd.cc:360]: I still don't understand why this is needed. A new wsrep transaction should always be started before THD accesses MDL or storage engine, so if a thd becomes a victim, it should always either:
* Have active transaction here, or
* Have been rolled back before BF thd enters here

Starting a wsrep transaction on behalf of another thread sounds wrong.

**Review by temeo — CHANGES_REQUESTED** [review]

File: sql/wsrep_thd.cc (hunk 2 above)
Comment (sjaakola) [reply sql/wsrep_thd.cc:360]: I don't remember why this was needed, wsrep_start_transaction call was planted by the original 8.0 merging code drop in commit: f0d1f7db6f598aa0408ee01928278dba539565af
I will put debug assert before this call, and we can see if it fires in our testing

**Comment by sjaakola** [comment]
Closing and submitting to MariaDB

**Pull request closed without merging**